### PR TITLE
Use LIEF for the ELF collector

### PR DESCRIPTION
**Pull request by romainthomas**

### CODE REVIEW

File: src/collectors/elf.cpp [hunk 1]
```diff
@@ -45,19 +46,14 @@ class ELFCollector : public Collector {
...
     static bool initialized;
     FILE* fd_;
-    Elf* elf_;
+    Elf_Binary_t* elf_binary_;
     boost::filesystem::path path_;
 
 public:
 
-  ELFCollector() : fd_(0), elf_(0) {
+  ELFCollector() : fd_(0), elf_binary_(NULL) {
```
Comment (serge-sans-paille) [inline src/collectors/elf.cpp]: NULL is a C-sim. in pre-c++11, the correct init is 0.

File: src/collectors/elf.cpp [hunk 2]
```diff
@@ -67,170 +63,138 @@ class ELFCollector : public Collector {
...
-    if (fd_ = fopen(input_file.c_str(), "r")) {
-      int fno = fileno(fd_);
-      if (elf_ = elf_begin(fno, ELF_C_READ, 0)) {
-        Elf_Kind ek = elf_kind(elf_);
-        if (ek != ELF_K_ELF) {
-          return false;
-        }
-      }
+
+    if (is_elf(input_file.string().c_str())) {
```
Comment (serge-sans-paille) [inline src/collectors/elf.cpp]: you could avoid the string() + c_str() repetition here.

File: src/collectors/elf.cpp [hunk 3]
```diff
@@ -67,170 +63,138 @@ class ELFCollector : public Collector {
...
-        break;
-      }
-    }
-    if (interp) {
-      std::string theinterp(interp);
-      delete[] interp;
-      return theinterp;
-    }
-    else
+    if (this->elf_binary_->interpreter != NULL) {
```
Comment (serge-sans-paille) [inline src/collectors/elf.cpp]: remove the ``!= NULL` here?

File: src/collectors/elf.cpp [hunk 4]
```diff
@@ -67,170 +63,138 @@ class ELFCollector : public Collector {
...
+      return this->elf_binary_->interpreter;
```
Comment (serge-sans-paille) [inline src/collectors/elf.cpp]: I **love** how much cleaner the code gets

**Comment by romainthomas** [comment]
@serge-sans-paille There are some bugs that are related to *Python Interface* that I can't understand. (cf Travis build).

**Comment by romainthomas** [comment]
Already done Serge: https://travis-ci.org/serge-sans-paille/binmap/jobs/307462288

**Pull request merged**